crates/red_knot_python_semantic/resources/mdtest/type_properties/is_disjoint_from.md

```diff
@@ -130,6 +130,18 @@ static_assert(not is_disjoint_from(Y, Z))
 static_assert(not is_disjoint_from(Intersection[X, Y], Z))
 static_assert(not is_disjoint_from(Intersection[X, Z], Y))
 static_assert(not is_disjoint_from(Intersection[Y, Z], X))
+
+# If one side has a positive element and the other side has a negative of that element, they are disjoint
```

Comment:
This is only true for fully-static types, since if we have dynamic types involved, `Any` on each side can represent an arbitrarily different type. In the simplest case, `Any` is not disjoint from `Not[Any]`: `Not[Any]` is indistinguishable from `Any`, and clearly `Any` is not disjoint from `Any`. Maybe we should add some more tests for this?

Reply:
Yeah I suspected in the comment I left above that Any was broken but then I couldn't construct a case to assert it. Found one now that I'll add, and replacing is_assignable_to with is_subtype_of fixes it

Reply:
is_subtype_of is also not a clean fix - regardless of which direction I choose one of these tests fails:

```
static_assert(not is_disjoint_from(Intersection[Any, X], Intersection[Any, Not[Y]]))
static_assert(not is_disjoint_from(Intersection[Any, Not[Y]], Intersection[Any, X]))

static_assert(is_disjoint_from(Intersection[int, Any], Not[int]))
static_assert(is_disjoint_from(Not[int], Intersection[int, Any]))
```

Need to think about this some more tomorrow